### PR TITLE
test(device): fuzz container codec round trip

### LINKED ISSUE #2582: test(device): add semantic round-trip property to container codec fuzzing
**What would you like to be added**:

Add a structure-aware fuzz test for the internal container-device annotation codec in `pkg/device/devices_test.go`.

The existing `FuzzDecodeContainerDevices` passes arbitrary strings to `DecodeContainerDevices`, but discards both returned values:

```go
_, _ = DecodeContainerDevices(str)
```

This is useful for panic detection, but it cannot detect a successful decode that returns incomplete or unexpected device data. Please retain that no-panic target and add a separate encoder-to-decoder property target, for example `FuzzEncodeDecodeContainerDevices`.

The new target should:

- Build `ContainerDevices` values from fuzzed UUID, type, memory, and core fields.
- Restrict UUID and type values to the delimiter-safe domain supported by the current positional wire format (`,`, `:`, and `;` are separators).
- Encode with `EncodeContainerDevices`.
- Decode the encoded annotation with `DecodeContainerDevices`.
- Assert that decoding returns no error and preserves every serialized field.
- Leave `Idx` and `CustomInfo` at their non-serialized defaults, since `EncodeContainerDevices` intentionally does not encode them.

This should remain a test-only change. It does not propose changing which externally supplied malformed annotations the decoder accepts.

Suggested verification:

```console
go test ./pkg/device -run '^$' -fuzz '^FuzzEncodeDecodeContainerDevices$' -fuzztime=30s
make verify
```

**Why is this needed**:

The current fuzz target from #2146 verifies only that arbitrary annotation input does not panic. The later node/pod fuzzing work in #2373 and #2376 follows the same no-panic pattern for decoder input, while adding deterministic round-trip coverage elsewhere. There is still no structure-aware fuzz property proving that annotations produced by `EncodeContainerDevices` can always be consumed by `DecodeContainerDevices` without losing the serialized allocation fields.

I noticed this gap after reporting #2507 and opening #2508. That change proposed rejecting malformed non-empty decoder segments, but #2508 was closed because this annotation is treated as an internal API and the preferred prevention was to test it fully before release. This proposal follows that direction directly: exercise the internal producer and consumer together without changing the parser contract.

This matters because a panic-only fuzz target would not catch the class of semantic failure discussed in #2507. Adding `"malformed"` to its corpus, for example, would still pass as long as the decoder did not panic. A round-trip oracle gives the fuzzer a concrete correctness property instead of only a process-safety property.

No GPU hardware or Kubernetes cluster is required.

**Anything else we need to know?**:

Related work reviewed before filing:

- #2146 added `FuzzDecodeContainerDevices`, with a no-panic invariant only.
- #2373 requested node/pod fuzz and round-trip coverage.
- #2376 implemented node/pod decoder fuzzing and deterministic node round-trip tests, but did not add a structure-aware container codec fuzz property.
- #1856 and #1857 improved decoder error propagation; they do not test encoder/decoder symmetry.
- #2507 and #2508 cover the malformed-segment behavior that led to this test-focused follow-up.
- #2514 and #2524 concern whitespace parsing and do not provide this property coverage.

I also searched current open and closed HAMi issues and pull requests for container codec fuzzing, semantic fuzz assertions, property tests, and `EncodeContainerDevices` round trips. I did not find an existing issue or pull request with this specific scope.

### PR DESCRIPTION
**Pull request by blackdragoon26**
**What type of PR is this?**
/kind failing-test

**What this PR does / why we need it**:

Adds a structure-aware fuzz test for the internal container-device annotation codec.

The existing `FuzzDecodeContainerDevices` verifies that arbitrary input does not panic, but it discards both the decoded result and error. Therefore, it cannot detect semantic regressions where an annotation is accepted but its serialized allocation fields are changed or lost.

The new `FuzzEncodeDecodeContainerDevices`:

- Generates UUID, device type, memory, and core values.
- Restricts UUID and type inputs to the delimiter-safe wire-format domain.
- Encodes the generated allocation with `EncodeContainerDevices`.
- Decodes it with `DecodeContainerDevices`.
- Verifies that decoding succeeds and preserves every serialized field.
- Covers empty, zero, negative, and `int32` boundary values.

The existing arbitrary-input/no-panic fuzzer remains unchanged. This PR is test-only and does not change annotation parsing behavior.

**Which issue(s) this PR fixes**:
Fixes #2582 

**Special notes for your reviewer**:

This follows the internal-API testing direction discussed after #2507 and #2508. It tests annotations generated by HAMi itself without reopening the malformed external-input parsing contract.

Validation performed on macOS:

- Focused fuzz corpus test passed.
- 30-second fuzz run passed with 268,803 executions.
- `go test ./pkg/device/...` passed.
- `make verify` passed in a clean checkout.
- `git diff --check` passed.

No GPU hardware or Kubernetes cluster is required.

**Does this PR introduce a user-facing change?**:
Nope, this is a test-only change.

AI Disclosure:
I used OpenAI Codex to inspect the related annotation-codec history, check for duplicate issues and pull requests, and assist with implementing and validating this focused fuzz test. I reviewed the complete change, understand the tested round-trip property, and ran all verification commands myself.

## Summary by CodeRabbit

* **Tests**
  * Added fuzz testing to verify reliable round-trip encoding and decoding of container-device data.
  * Added coverage for preserving valid input values without errors.

### CODE REVIEW

File: pkg/device/devices_test.go [hunk 1]
```diff
@@ -1722,6 +1723,28 @@ func FuzzDecodeContainerDevices(f *testing.F) {
 	})
 }
 
+func FuzzEncodeDecodeContainerDevices(f *testing.F) {
```
Comment (mesutoezdil) [inline pkg/device/devices_test.go:1726]: containerdevice also has idx and a pending slots field from pr #2623. why fuzz only uuid, type, usedmem, usedcores and not those two?

Comment (blackdragoon26) [reply pkg/device/devices_test.go:1726]: Because the current annotation contract as per this merged PR[#2185](https://github.com/Project-HAMi/HAMi/pull/2185) and the implementation done  by [EncodeContainerDevices](https://github.com/Project-HAMi/HAMi/blob/master/pkg/device/devices.go#L448-L452), encodes only, UUID, type, memory, and cores.
Plus, [Idx is explicitly marked as non-encoded](https://github.com/Project-HAMi/HAMi/blob/master/pkg/device/devices.go#L209-L216), so including it in a DeepEqual round trip would fail by design.

Comment (blackdragoon26) [reply pkg/device/devices_test.go:1726]: slots is not on master and #2623 keeps it non-encoded, so i have kept it outside this round-trip property.

**Review by archlitchi — APPROVED** [review]
/lgtm
